### ISSUE TITLE
Feature: Keep a backup, rollback  

### ISSUE THREAD

**Issue by staberas** [ozgrozer/ai-renamer#8]
Keep a backup and a rollback feature in case a picture gets renamed by accident.

**Comment by ozgrozer** [comment]
Yeah maybe a --backup flag might be useful but what type of accident are we talking about?

**Comment by staberas** [comment]
In a folder that i have a bunch of unsorted memes and various images without clear title, some of my photos do have  the creator-person in the picture name is being changed to something descriptive but not helpful. 
for example -> " julia-lepetit.png to woman-posing-infront-graffiti.png " yes the title is correct but i like to have the name of the person in the pic i used . Thats why some json/txt backup of the changes might be needed

**Comment by ozgrozer** [comment]
I don't think that's an issue. The script does what it has do. Instead of a backup and rollback, maybe a confirmation message would be more suitable.

**Comment by staberas** [comment]
Confirmation before it starts, and a log of name changes (as a backup), that would be 
the log should log folder the change happened and the changes example of log/backup file

```
Folder C:\folderwithfiles
structures.jpg to lor-gradient-applied-to-image.jpg
discernible-text.jpg to ism-and-its-impact-on-society.jpg
feel-free-to-ask.jpg to photo-background.jpg
unique-style-and-interests.jpg to ate-about-the-story-behind-it.jpg
as-per-your-request.jpeg to lock-wall-with-graffiti-on-it.jpeg
Folder C:\folderwithfiles\subfolder
otherimage.jpg to ate-about-the-story-behind-it.jpg
iwanttorename.jpeg to lock-wall-with-graffiti-on-it.jpeg
```

**Comment by ozgrozer** [comment]
Let's make this issue open. I'll implement the confirmation feature if it's asked some more.

**Comment by darrenpierre90** [comment]
> Confirmation before it starts, and a log of name changes (as a backup), the log should log folder-filename changes that happened. Example of log/backup file
> 
> ```
> Folder C:\folderwithfiles
> structures.jpg to lor-gradient-applied-to-image.jpg
> discernible-text.jpg to ism-and-its-impact-on-society.jpg
> feel-free-to-ask.jpg to photo-background.jpg
> unique-style-and-interests.jpg to ate-about-the-story-behind-it.jpg
> as-per-your-request.jpeg to lock-wall-with-graffiti-on-it.jpeg
> Folder C:\folderwithfiles\subfolder
> otherimage.jpg to ate-about-the-story-behind-it.jpg
> iwanttorename.jpeg to lock-wall-with-graffiti-on-it.jpeg
> ```

I would love a confirmation message before it starts as I'm not quite happy with the results .

**Comment by XGhozt** [comment]
Or perhaps instead of renaming, we have an option to prepend or append the new description?

**Comment by chla-conscia** [comment]
I'm glad I tried it on some test-files first.
I wasnt satisfied with the renaming.
so now I dont know if I wanna make the big bulk.

So a --backup flag would be very appreciate

**Comment by unstatusthequo** [comment]
> Or perhaps instead of renaming, we have an option to prepend or append the new description?

Yes, I second this. An option to retain the original filename in part would be nice. So you can decide to add the new filename as a prefix or suffix to the original filename. Or remove it, which is what we have now.

**Comment by IceAsteroid** [comment]
I also need this, it's gonna be really useful!

**Comment by barakplasma** [comment]
Design proposal: could be implemented with an SQLite file database using [node:sqlite](https://nodejs.org/docs/latest-v22.x/api/sqlite.html) and a table with the schema

|unix-time|old_absolute_path|new_absolute_path|
|----|----|----|
|1746903002|/2024fjsfs9f.jpg|/three-dogs-playing.jpg|
|1746903150|/three-dogs-playing.jpg|/playful-dog-park.jpg|

a user could play a rename backwards with something like `ai-renamer --undo playful-dog-park.jpg` which would lookup the latest new_absolute path matching the given path and rename that path back to the old_absolute_path . If they wanted another `undo`, just lookup the next one. If there are duplicates of the same path somehow, the older path keeps its name and the newer path gets the current unix-time appended to the end of the name.

**Comment by barakplasma** [comment]
> Confirmation before it starts, and a log of name changes (as a backup), the log should log folder-filename changes that happened. Example of log/backup file
> 
> ```
> Folder C:\folderwithfiles
> structures.jpg to lor-gradient-applied-to-image.jpg
> discernible-text.jpg to ism-and-its-impact-on-society.jpg
> feel-free-to-ask.jpg to photo-background.jpg
> unique-style-and-interests.jpg to ate-about-the-story-behind-it.jpg
> as-per-your-request.jpeg to lock-wall-with-graffiti-on-it.jpeg
> Folder C:\folderwithfiles\subfolder
> otherimage.jpg to ate-about-the-story-behind-it.jpg
> iwanttorename.jpeg to lock-wall-with-graffiti-on-it.jpeg
> ```

I like this but I would recommend adding timestamps and maybe as markdown or csv format to make it easier to edit in excel / a text editor or IDE / or with another AI prompt ("given this markdown file, translate the names to spanish")

|datetime|old|new|
|---|---|---|
|1746904287|structures.jpg|lor-gradient-applied-to-image.jpg|


```csv
datetime,old,new
1746904287,structures.jpg,lor-gradient-applied-to-image.jpg,
```

**Comment by teolemon** [comment]
Another option is to generate suggestions, and have the user approve or refuse them, using big YES / NO buttons and/or keyboard shortcuts
